lite2/client.go

```diff
@@ -604,10 +624,10 @@ func (c *Client) sequence(
 		err = VerifyAdjacent(c.chainID, trustedHeader, interimHeader, interimVals,
 			c.trustingPeriod, now)
 		if err != nil {
-			err = errors.Wrapf(err, "verify adjacent from #%d to #%d failed",
-				trustedHeader.Height, interimHeader.Height)
+			err = fmt.Errorf("verify adjacent from #%d to #%d failed: %w",
+				trustedHeader.Height, interimHeader.Height, err)
 
-			switch errors.Cause(err).(type) {
+			switch (err).(type) {
```

Comment:
```suggestion
			switch errors.Unwrap(err).(type) {
```

Reply:
I can change that but can you tell me what the difference between errors.Unwrap, errors.Cause and err.(type) which I have seen all used in the same way

Reply:
Cause - https://github.com/pkg/errors#retrieving-the-cause-of-an-error
Unwrap is the same ^